x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/DenseVectorFieldTypeIT.java

```diff
@@ -0,0 +1,177 @@
+        this.indexType = indexType;
+        this.index = index;
+        this.synthetic = synthetic;
+    }
+
+    private final Map<Integer, List<Float>> indexedVectors = new HashMap<>();
+
+    public void testRetrieveFieldType() {
+        var query = """
+            FROM test
+            """;
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "vector"));
+            assertColumnTypes(resp.columns(), List.of("integer", "dense_vector"));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testRetrieveTopNDenseVectorFieldData() {
+        var query = """
+                FROM test
+                | KEEP id, vector
+                | SORT id ASC
+            """;
+
+        try (var resp = run(query)) {
+            List<List<Object>> valuesList = EsqlTestUtils.getValuesList(resp);
+            indexedVectors.forEach((id, vector) -> {
+                var values = valuesList.get(id);
+                assertEquals(id, values.get(0));
+                List<Float> vectors = (List<Float>) values.get(1);
+                assertNotNull(vectors);
+                assertEquals(vector.size(), vectors.size());
+                for (int i = 0; i < vector.size(); i++) {
+                    assertEquals(vector.get(i), vectors.get(i), 0F);
+                }
+            });
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testRetrieveDenseVectorFieldData() {
+        var query = """
+            FROM test
+            | KEEP id, vector
+            """;
+
+        try (var resp = run(query)) {
+            List<List<Object>> valuesList = EsqlTestUtils.getValuesList(resp);
+            assertEquals(valuesList.size(), indexedVectors.size());
+            valuesList.forEach(value -> {
+                ;
+                assertEquals(2, value.size());
+                Integer id = (Integer) value.get(0);
+                List<Float> vector = (List<Float>) value.get(1);
+                assertNotNull(vector);
+                List<Float> expectedVector = indexedVectors.get(id);
+                assertNotNull(expectedVector);
+                for (int i = 0; i < vector.size(); i++) {
+                    assertEquals(expectedVector.get(i), vector.get(i), 0F);
+                }
+            });
+        }
+    }
+
+    @Before
+    public void setup() throws IOException {
+        var indexName = "test";
+        var client = client().admin().indices();
+        XContentBuilder mapping = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject("id")
+            .field("type", "integer")
+            .endObject()
+            .startObject("vector")
+            .field("type", "dense_vector")
+            .field("index", index);
+        if (index) {
+            mapping.field("similarity", "l2_norm");
```

Comment:
nit: is it possible to randomize the similarity option? otherwise we could leave this completely out, since it optional?